[docs] Iterate on the job ad for React engineer in Core (#32900)
mui/material-ui: docs/src/pages/careers/react-engineer-core.md

- **You've maintained an active repository before**.
  Maybe you've helped maintain a popular open-source repository, or perhaps you've worked on internal repositories that saw contributions from multiple teams.
  Previous experience with highly active repository workflows is a definite plus for this role.
- **You have contributed code to MUI before**.
  A history of contributing to MUI would be a definite plus.
- **You have experience with accessibility**.

>   Maybe you were responsible or contributed to making sure that the design system/application at your company is a11y compliant.

Reviewer: ```suggestion
  Maybe you've been involved in bringing a11y compliance to your company's design system or application.
```

Reviewer: Maybe consider adding a second line here for extra emphasis? (Assuming this is true, which I think it is 😅)

```suggestion
  Maybe you've been involved in bringing a11y compliance to your company's design system or application.
  We would greatly benefit from having someone on the team who can take ownership of this.
```
